File: backend/MiniBilge.Application/DTOs/Profile/ParentProfileDto.cs
namespace MiniBilge.Application.DTOs.Profile;

public class ParentProfileDto
{
    public Guid Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string? PhoneNumber { get; set; }
    public int ChildrenCount { get; set; }
}

File: mobile/minibilge_mobile/lib/features/auth/models/parent_profile_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'parent_profile_dto.freezed.dart';
part 'parent_profile_dto.g.dart';

@freezed
class ParentProfileDto with _$ParentProfileDto {
  const factory ParentProfileDto({
    @JsonKey(name: 'Id') required String id,
    @JsonKey(name: 'FirstName') required String firstName,
    @JsonKey(name: 'LastName') required String lastName,
    @JsonKey(name: 'PhoneNumber') String? phoneNumber,
    @JsonKey(name: 'ChildrenCount') required int childrenCount,
  }) = _ParentProfileDto;

  factory ParentProfileDto.fromJson(Map<String, dynamic> json) =>
      _$ParentProfileDtoFromJson(json);
}
